src/Compilers/CSharp/Portable/Symbols/ParameterSymbol.cs

```diff
@@ -385,6 +385,56 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
 
         internal abstract bool IsCallerMemberName { get; }
 
+        /// <summary>
+        /// If the method returns false, then we can infer that the argument used for this parameter was not null.
+        /// </summary>
+        internal abstract bool NotNullWhenFalse { get; }
+
+        /// <summary>
+        /// When the method returns, then we can infer that the argument used for this parameter was not null.
+        /// </summary>
+        internal abstract bool EnsuresNotNull { get; }
+
+        protected bool HasExtraAttribute(AttributeDescription description)
+        {
+            ParameterSymbol originalParameter = this.OriginalDefinition;
+            var containingMethod = originalParameter.ContainingSymbol as MethodSymbol;
+
+            if (containingMethod is null)
+            {
+                return false;
+            }
+
+            string key = ExtraAnnotations.MakeMethodKey(containingMethod);
+
+            // index 0 is used for return type
+            // `this` parameter is at index 1
+            // other parameters follow
+            int index = this.Ordinal + (containingMethod.IsExtensionMethod ? 2 : 1);
```

Comment:
Isn't the first parameter for an extension method at the same index as the first parameter of any static method?

Reply:
I couldn't find a good example of a BCL extension method that would need extra annotations. I'll block this scenario for now.

---
In reply to: [187092735](https://github.com/dotnet/roslyn/pull/26656#discussion_r187092735) [](ancestors = 187092735)